### ISSUE TITLE
Prepare diagrams for multiple database schema

### ISSUE THREAD

**Issue by rafalkasa** [schemaspy/schemaspy#1]

**Comment by jotpe** [comment]
Yes, would be perfect to see everything inside a database, not only inside one schema (you have to know before).

**Comment by rafalkasa** [comment]
Hi 

I believe that such change in how SchemaSpy is working right now will be very usable. 

For some users current solution is good, so for this reason I plan to add extra parameter to decide how SchemaSpy will be analysis the database.

In new method Instead of analysis schema by schema SchemaSpy should analysis all schemas and prepare common diagrams.

**Comment by jotpe** [comment]
Can I support you, somehow?

**Comment by rafalkasa** [comment]
This issue reference to SchemaSpy discussion from sourceforge https://sourceforge.net/p/schemaspy/discussion/462849/thread/197be2a4/

**Comment by Pinimo** [comment]
:+1:
I think this enhancement would indeed be great : I would like an option flag enabling cross-schema foreign key constraints to appear on a general graph with tables from all schemas (even if it might become quite a mess).

+ suggestion : in this option, schemas could be differentiated with colors.

**Comment by rafalkasa** [comment]
It's look like more and more people start to be interesting about this feature. This mean that I should start finding the best approach to invent in SchemaSpy diagrams for multiple database schema.

Thank you @Pinimo for your vote and comment

**Comment by Pinimo** [comment]
Thank you @rafalkasa. One more suggestion: perhaps an easy way of implementing this could involve :
- generating a fake (empty) duplicate of the DB at computation time (copying table headers and constraints but not their contents -- dunno if it's possible), 
- then applyinfg SQL `alter table myTable set schema temporarySchema;` queries on all tables
- this way one would already have the correct diagrams etc
- only the colors (or some other schema indication) would be left to take care of, as well as replacing the row counts (as we did an empty copy of the DB).

**Comment by npetzall** [comment]
@rafalkasa @wkasa since the two commits above where do we stand what is status. What is left to do?

**Comment by richardmangueira** [comment]
Hello guys,
First of all I would like to thank the schemaspy creator, congratulations is fabulous.
Any solutions came up?

**Comment by npetzall** [comment]
#204 might be a step in the right direction, since we would do the complete dbms and as such we can create a diagrams based on dbms, catalog and schema, we could have three depths if they are all available.

**Comment by exetico** [comment]
Are there a way to run this for multiple DB's? In a mssql instance, I'm normally having mulitple DB's. And it would be nice to have it generated for all DB's at once, instead of having to execute it manually for each DB on the server.
